### PR TITLE
Add Address display test

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

Adds a test to make sure all address display implementations are.. somewhat similar.

Q: Do we want random AddressCapabilities, weights and thresholds?

### CODE REVIEW

File: sdk/src/types/block/address/mod.rs [hunk 1]
```diff
@@ -183,3 +183,22 @@ impl<T: Into<Address>> ToBech32Ext for T {
         Bech32Address::new(hrp.convert_unchecked(), self)
     }
 }
+
+mod test {
+    use crate::types::block::address::Address;
+    use crate::types::block::rand::address::*;
+
+    #[tokio::test]
+    async fn address_display_similar() {
```
Comment (thibault-martinez) [inline sdk/src/types/block/address/mod.rs]: Why async?

File: sdk/src/types/block/address/mod.rs [hunk 2]
```diff
@@ -183,3 +183,22 @@ impl<T: Into<Address>> ToBech32Ext for T {
         Bech32Address::new(hrp.convert_unchecked(), self)
     }
 }
+
+mod test {
```
Comment (thibault-martinez) [inline sdk/src/types/block/address/mod.rs]: Move it to the tests folder

File: sdk/src/types/block/rand/address.rs [hunk 3]
```diff
@@ -27,6 +27,26 @@ pub fn rand_anchor_address() -> AnchorAddress {
...
+pub fn rand_restricted_address() -> RestrictedAddress {
+    RestrictedAddress::new(rand_address()).unwrap().with_allowed_capabilities(AddressCapabilities::all())
+}
+
+/// Generates a random Multi address.
+pub fn rand_multi_address() -> MultiAddress {
+    MultiAddress::new([rand_weighted_address()], 1).unwrap()
+}
+
+/// Generates a random Multi address.
```
Comment (thibault-martinez) [inline sdk/src/types/block/rand/address.rs]: not multi

File: sdk/src/types/block/address/mod.rs [hunk 4]
```diff
@@ -183,3 +183,22 @@ impl<T: Into<Address>> ToBech32Ext for T {
...
+    use crate::types::block::address::Address;
+    use crate::types::block::rand::address::*;
```
Comment (thibault-martinez) [inline sdk/src/types/block/address/mod.rs]: merge

File: sdk/src/types/block/address/mod.rs (hunk 1 above)
Comment (kwek20) [reply sdk/src/types/block/address/mod.rs]: copy/paste

File: sdk/src/types/block/address/mod.rs (hunk 2 above)
Comment (kwek20) [reply sdk/src/types/block/address/mod.rs]: im not sure, needs random feature, weh ave that in internal always right?

Comment (thibault-martinez) [reply sdk/src/types/block/address/mod.rs]: No it's public, most tests in tests use random features

File: sdk/src/types/block/address/mod.rs [hunk 5]
```diff
@@ -183,3 +183,22 @@ impl<T: Into<Address>> ToBech32Ext for T {
         Bech32Address::new(hrp.convert_unchecked(), self)
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::types::block::{address::Address, rand::address::*};
+
+    #[tokio::test]
```
Comment (thibault-martinez) [inline sdk/src/types/block/address/mod.rs]: no need for tokio then :trollface:

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**